src/dbnode/storage/cleanup.go

```diff
@@ -301,26 +297,75 @@ func (m *cleanupManager) commitLogTimes(t time.Time) (time.Time, []time.Time, er
 	// are only retained for a period of 1-2 days (at most), after we which we'd live we with the
 	// data loss.
 
-	candidateTimes := timesInRange(earliest, latest, blockSize)
+	files, err := m.commitLogFilesFn(m.opts.CommitLogOptions())
+	if err != nil {
+		return nil, err
+	}
 	namespaces, err := m.database.GetOwnedNamespaces()
 	if err != nil {
-		return time.Time{}, nil, err
+		return nil, err
 	}
-	cleanupTimes := filterTimes(candidateTimes, func(t time.Time) bool {
+
+	var outerErr error
+	filesToCleanup := filterCommitLogFiles(files, func(start time.Time, duration time.Duration) bool {
+		if outerErr != nil {
+			return false
+		}
+
+		if start.Before(earliest) {
+			// Safe to clean up expired files.
+			// TODO(rartoul): Now that we have commit log compaction via snapshot files we would like
+			// to remove the concept of commit log retention so that users cannot accidentally
+			// configure M3DB in such a way that it loses data (commit log retention < block size).
+			return true
+		}
+
 		for _, ns := range namespaces {
-			ropts := ns.Options().RetentionOptions()
-			start, end := commitLogNamespaceBlockTimes(t, blockSize, ropts)
-			if ns.NeedsFlush(start, end) {
+			var (
+				ropts                      = ns.Options().RetentionOptions()
+				nsBlocksStart, nsBlocksEnd = commitLogNamespaceBlockTimes(start, duration, ropts)
+				needsFlush                 = ns.NeedsFlush(nsBlocksStart, nsBlocksEnd)
```

Comment:
Hm, I wonder if we should change these type of functions to instead of taking `start, end time.Time` we should be using a single `xtime.Range` arg?

Reply:
not sure how much it buys us as all the helper functions take a start/end as well and we're not trying to calculate overlap or anything like that

Reply:
Yeah, just more that the name of the data structure represents the variable itself better and its a single var that can then be passed around.  I'm not bullish though, we can revisit this later.

Reply:
cool lets leave this out in the interest of time for now